presentation/authentication/src/main/java/com/giraffe/authentication/validation/ValidationError.kt

```diff
@@ -0,0 +1,18 @@
+package com.giraffe.authentication.validation
+
+import android.content.Context
+import com.giraffe.authentication.R
+
+interface ValidationError
+
+object InvalidEmail : ValidationError
+object InvalidPassword : ValidationError
+
+
+fun ValidationError.getMessage(context: Context): String {
+    return when (this) {
+        InvalidEmail -> context.getString(R.string.invalid_email_address)
+        InvalidPassword -> context.getString(R.string.invalid_password_minimum_8_characters)
+        else -> "Unknown Error"
+    }
+}
```

Comment:
using view context in function can cause memory leak, so we should avoid that. change this function to return stringId and get string stringRes compose function:
 
```suggestion
@StringRes
fun ValidationError.getMessage(): Int {
    return when (this) {
        InvalidEmail -> R.string.invalid_email_address
        InvalidPassword -> R.string.invalid_password_minimum_8_characters
        else -> R.string.unkown_error //make this string in resources too
    }
}
```

Reply:
why memory leak? 
can you explain to me more?